src/Http/GenericRouter.php

```diff
@@ -172,4 +160,68 @@ private function createResponse(Response|View $input): Response
 
         return $input;
     }
+
+    private function matchStaticRoute(Request $request): ?MatchedRoute
+    {
+        $staticRoute = $this->getRoutes()[$request->method->value][$request->getPath()] ?? null;
+
+        if ($staticRoute === null) {
+            return null;
+        }
+
+        // TODO: why do we need to check this?
+        if ($staticRoute->isDynamic) {
+            return null;
+        }
+
+        return new MatchedRoute($staticRoute, []);
+    }
+
+    private function matchDynamicRoute(Request $request): ?MatchedRoute
+    {
+        // If there are no routes for the given request method, we immediately stop
+        $routesForMethod = $this->getRoutes()[$request->method->value] ?? null;
+        if ($routesForMethod === null) {
+            return null;
+        }
+
+        // Next, we'll build one big regex to match the correct route
+        // See https://github.com/tempestphp/tempest-framework/pull/175 for the details
+
+        /** @var \Tempest\Http\Route[] $routesForMethod */
+        $routesForMethod = array_values($routesForMethod);
+        $combinedMatchingRegex = "#^(?|";
+
+        foreach ($routesForMethod as $routeIndex => $route) {
+            if (! $route->isDynamic) {
+                continue; // TODO: why this check?
```

Comment:
Also the same question here: we can probably get rid of this check when we split dynamic and static routes?